src/rocksdb_options.rs

```diff
@@ -158,7 +158,7 @@ impl UnsafeSnap {
 
 pub struct ReadOptions {
     inner: *mut DBReadOptions,
-    upper_bound: Vec<u8>,
+    upper_bound: Option<Vec<u8>>,
```

Comment:
@siddontang no
@BusyJay this is the optimization for a normally-empty Vec. not ridiculous or something.